lib/Doctrine/DBAL/Platforms/AbstractPlatform.php

```diff
@@ -1005,254 +879,169 @@ public function getCosExpression($value)
-     * @param string $date
-     * @param int    $weeks
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateAddWeeksExpression($date, $weeks)
+    public function getDateAddWeeksExpression(string $date, string $weeks) : string
     {
         return $this->getDateArithmeticIntervalExpression($date, '+', $weeks, DateIntervalUnit::WEEK);
     }
 
     /**
      * Returns the SQL to subtract the number of given weeks from a date.
      *
-     * @param string $date
-     * @param int    $weeks
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateSubWeeksExpression($date, $weeks)
+    public function getDateSubWeeksExpression(string $date, string $weeks) : string
     {
         return $this->getDateArithmeticIntervalExpression($date, '-', $weeks, DateIntervalUnit::WEEK);
     }
 
     /**
      * Returns the SQL to add the number of given months to a date.
      *
-     * @param string $date
-     * @param int    $months
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateAddMonthExpression($date, $months)
+    public function getDateAddMonthExpression(string $date, string $months) : string
     {
         return $this->getDateArithmeticIntervalExpression($date, '+', $months, DateIntervalUnit::MONTH);
     }
 
     /**
      * Returns the SQL to subtract the number of given months to a date.
      *
-     * @param string $date
-     * @param int    $months
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateSubMonthExpression($date, $months)
+    public function getDateSubMonthExpression(string $date, string $months) : string
     {
         return $this->getDateArithmeticIntervalExpression($date, '-', $months, DateIntervalUnit::MONTH);
     }
 
     /**
      * Returns the SQL to add the number of given quarters to a date.
      *
-     * @param string $date
-     * @param int    $quarters
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateAddQuartersExpression($date, $quarters)
+    public function getDateAddQuartersExpression(string $date, string $quarters) : string
     {
         return $this->getDateArithmeticIntervalExpression($date, '+', $quarters, DateIntervalUnit::QUARTER);
     }
 
     /**
      * Returns the SQL to subtract the number of given quarters from a date.
      *
-     * @param string $date
-     * @param int    $quarters
-     *
-     * @return string
-     *
      * @throws DBALException If not supported on this platform.
      */
-    public function getDateSubQuartersExpression($date, $quarters)
+    public function getDateSubQuartersExpression(string $date, string $quarters) : string
```

Comment:
`$quarters` should be `int` according to phpDoc so this is a BC break.

Reply:
Addressed in #3498.